### PR TITLE
Adds support for DataCite JSON API.

### LINKED ISSUE #599: Use Datacite for DOIs
### Descriptive summary

The DOI feature needs to be updated to use Datacite as a provide instead of EZID.

The first step for this task is to review the approach that has been discussed in the past.  See @hortongn 

Existing DOIs will need to be migrated to Datacite as well.  That needs to be done at the same time this enhancement is deployed.

### PR DESCRIPTION
**Pull request by chuckgreenman**
Fixes #599  ; refs #755 

EZID is shutting down service to non California based institutions which made it necessary to transition to another DOI provider.  Several changes were made to the Scholar application but our fork of hydra-remote_identifier also contains changes that were necessary for the migration.

This revision replicates the functionality that is present on production, here are some known issues that we should follow up with:

- When a work is embargoed and has a minted DOI, the DOI's status does not get updated from registered to published.
- When a work is updated, the metadata in our provider is not updated.

We also ran into a problem where `identifier_url` in the database was still pointing at the EZID service, this is tracked with issue #755 and has been resolved.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,13 +8,13 @@ git_source(:github) do |repo_name|
...
 gem 'browse-everything', github: 'uclibs/browse-everything', branch: 'master'
-gem 'hydra-remote_identifier', github: 'uclibs/hydra-remote_identifier', branch: 'setting-status'
+gem 'hydra-remote_identifier', github: 'uclibs/hydra-remote_identifier', branch: 'datacite-wip-no-vcr'
 gem 'kaltura', '0.1.1'
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 5.1.6.1'
 # Use sqlite3 as the database for Active Record
-gem 'sqlite3'
+gem 'sqlite3', '1.3.13'
```
[the remark this answers is not in the document]
Comment (chuckgreenman) [reply Gemfile]: Pinning the gem makes active record happy, without explicitly providing a version number we get the following error message on attempting to start the rails server:
`Specified 'sqlite3' for database adapter, but the gem is not loaded. Add `gem 'sqlite3'` to your Gemfile (and ensure its version is at the minimum required by ActiveRecord). (Gem::LoadError)`

File: app/jobs/identifier_delete_job.rb [hunk 2]
```diff
@@ -2,23 +2,35 @@
...
+      RestClient.delete post_uri(identifier_uri)
+    else
+      RestClient.put(post_uri(identifier_uri), "{\n  \"data\": {\n    \"type\": \"dois\",\n    \"attributes\": {\n      \"event\": \"hide\",\n   \"url\": \"https://datacite.org/invalid.html\"\n    }\n  }\n}", content_type: :json)
+      # datacite_resource(identifier_uri).put("{\n  \"data\": {\n    \"type\": \"dois\",\n    \"attributes\": {\n      \"event\": \"hide\",\n   \"url\": \"https://datacite.org/invalid.html\"\n    }\n  }\n}", content_type: :json)
+    end
   end
 
   private
 
+    def datacite_resource(identifier_uri)
```
[the remark this answers is not in the document]
Comment (chuckgreenman) [reply app/jobs/identifier_delete_job.rb]: My vote would be to keep it, it is used by get_state()

File: spec/features/doi_validation_spec.rb [hunk 3]
```diff
@@ -1,12 +1,13 @@
 # frozen_string_literal: true
 require 'rails_helper'
+require 'byebug'
```
[the remark this answers is not in the document]
Comment (chuckgreenman) [reply spec/features/doi_validation_spec.rb]: Removed

File: spec/support/shared/is_remotely_identifiable_by_doi.rb [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require 'byebug'
```
[the remark this answers is not in the document]
Comment (chuckgreenman) [reply spec/support/shared/is_remotely_identifiable_by_doi.rb]: Removed

File: Gemfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (scherztc) [reply Gemfile]: +1

File: app/jobs/identifier_delete_job.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (scherztc) [reply app/jobs/identifier_delete_job.rb]: 👍

**Comment by scherztc** [comment]
@hortongn : Make sure you have an environment.local file with the test credentials if you want to QA it in development.

**Pull request merged**